src/gov/nist/core/LexerCore.java

```diff
@@ -411,14 +460,31 @@ public String ttoken() {
             return null;
         }
     }
+    
+    public String tIpv6address() {
+        int startIdx = ptr;
+        try {
+            while (hasMoreChars()) {
```

Comment:
No, beside checking the ']' is at the end of address , I also want to check that there is a valid IPV6 address character or not.